qualtran/bloqs/mcmt/ctrl_spec_and.py

```diff
@@ -224,3 +226,111 @@ def _ctrl_on_multiple_values() -> CtrlSpecAnd:
     bloq_cls=CtrlSpecAnd,
     examples=(_ctrl_on_int, _ctrl_on_bits, _ctrl_on_nd_bits, _ctrl_on_multiple_values),
 )
+
+
+@frozen
+class ControlledViaAnd(Bloq):
```

Comment:
@mpharrigan Do you think the Bloq should derive from `Controlled` instead of `Bloq` so we can reuse some of the infrastructure like `wire_symbol`, `classical_action`, getting `ctrl_reg_names` and `ctrl_regs` ?

Reply:
Oh I didn't think of this, I really like the idea! If it doesn't break anything, I'd prefer going with this, makes the interface cleaner.

Reply:
inherited from Controlled